packages/core/src/transport/httpRequest.ts

```diff
@@ -14,7 +15,16 @@ export class HttpRequest {
   constructor(private endpointUrl: string, private bytesLimit: number, private withBatchTime: boolean = false) {}
 
   send(data: string | FormData, size: number) {
-    const url = this.withBatchTime ? addBatchTime(this.endpointUrl) : this.endpointUrl
+    let url = this.endpointUrl
+
+    if (isEndpointIntakeV2(url)) {
```

Comment:
Checking if we are using the intake v2 at this level is a bit unexpected.

What would be the impact of always adding the request id? Does the "intake v1" ignore it or not?

Reply:
The intake ignores it, but it pass it along to rum backend. We have the issue with (also with intake v2) of the `dd-request-id` being in the JSON object section of the frontend. Once it's fixed, that should not be an issue.

Reply:
So, maybe we could wait until the issue is fixed? `intake v2` is still behind a flag anyway.

Reply:
what about handling that with a `withRequestId` constructor param depending on intakeApiVersion config + FF?